py3status/py3.py

```diff
@@ -933,7 +933,7 @@ def command_run(self, command):
             msg = 'Command `{cmd}` {error}'.format(cmd=pretty_cmd, error=e.errno)
             raise exceptions.CommandError(msg, error_code=e.errno)
 
-    def command_output(self, command, shell=False, capture_stderr=False, localized=False):
+    def command_output(self, command, shell=False, capture_stderr=False, localized=False, ignore_stderr=False):
```

Comment:
In many modules, we use this.
```python
        try:
            self.py3.command_output('REPLACE_ME')
        except self.py3.CommandError as ce:
            return ce.output
```

Reply:
To be honest, maybe the default here should be changed to True as well.

I'm not sure what you wanted to showcase with your example?

Reply:
An example above is a clear indicator of an `out to stderr` text.

If we use `ignore_stderr=False`, we can't tell easily which std throws an output. In the long run, we will end up assuming it's always `stderr` rather than seeing it can be cycled between both.

Also, per module error messages sometimes have to be fixed first (eg removing prefix, omitting empty lines, or picking certain line, etc). If we do have to fix them, then what is the point of having this argument in first place?

Simply put, you could be adding an argument that nobody would use... And it's more about having an ability to handle the output better in all modules than to simplify something inside `py3` so you can kill off few lines in your new module.

Reply:
See `arch_updates` module. Right now, it uses `subprocess.CalledProcessError` and should be converted to `self.py3.command_output()` eventually. Anyway, we need `try/except` here so we can return `None` when users can't get an output.

Users can't get an output because they're not connected to Internet. With `None`, users can make their `arch_updates` disappear (eg `[{pacman}/{aur}]`) or display something else.

Reply:
`external_script` appears to not have something to deal with `stderr` outputs.
```diff
diff --git a/py3status/modules/external_script.py b/py3status/modules/external_script.py
index 6bc231cb..969e0c82 100644
--- a/py3status/modules/external_script.py
+++ b/py3status/modules/external_script.py
@@ -54,6 +54,7 @@ class Py3status:
     localize = True
     script_path = None
     strip_output = False
+    ignore_stderr = False
 
     def post_config_hook(self):
         if not self.script_path:
@@ -73,7 +74,8 @@ class Py3status:
         except self.py3.CommandError as e:
             # something went wrong show error to user
             output = e.output or e.error
-            self.py3.error(output)
+            if not self.ignore_stderr:
+                self.py3.error(output)
 
         if output_lines:
             output_text = output_lines[0]

```

Reply:
Repeat for this conversation: I did something [bad](https://github.com/lasers/py3status/tree/newmodule-external_script). First draft.

Reply:
Maybe I'm still missing the point, but the patch above has one flaw: it ignores ALL errors, even true errors. I do want to execute self.py3.error() if command exited with a non-zero exit code, because non-zero exit code is what normally indicates an error. But if command exited with zero exit code, then I don't want to execute self.py3.error. 

I'll have another look at arch_updates and your new module in the evening, I'm currently replying from phone.

Reply:
@lasers I'm looking at `arch_updates`, that case is special because `cower` is broken`, `cower --update` returns non-zero status code on success. This is abnormal. We can change the command to be `cower --update || true`, this will always return zero exit code and so we can process the output as normal.